scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala

```diff
@@ -76,7 +77,31 @@ class GitOpsImpl(private[util] val workingDirectory: AbsoluteFile)
     )
     for {
       root <- rootDir.toSeq
-      path <- exec(cmd).toOption.toSeq.flatten
+      path <- exec(cmd).getOrElse(Seq.empty)
     } yield AbsoluteFile.fromFile(new File(path), root)
   }
+
+  override def status: Seq[AbsoluteFile] = {
+    val cmd = Seq(
+      "git",
+      "status",
+      "--porcelain"
+    )
+    for {
+      root <- rootDir.toSeq
+      statusLine <- exec(cmd).getOrElse(Seq.empty)
+      file <- getFileFromGitStatusLine(statusLine).toSeq
+    } yield AbsoluteFile.fromFile(file, root)
+  }.filter(_.exists)
+
+  private final val statusCodesSubstringLength = 3
+  private final val moveStatusArrowDelimiter = "-> "
+
+  private def getFileFromGitStatusLine(s: String): Option[File] =
+    s.substring(statusCodesSubstringLength)
+      .split(moveStatusArrowDelimiter) match {
```

Comment:
In my environment (details below), this method sometimes failed to extract the file correctly.

<details>

```
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"

$ git version
git version 2.21.0
```

</details>

Because, in some cases, the status line shows something like following (status codes occupy only one column), and, as a result, `s.substring(3)` returns `calafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala`, instead of `scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala`.

```
M scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala
```

I'm not sure why this is happening ... (sry)

---

What do you think about extracting the file names by this way instead?

```scala
s.trim.split(" ").tail.mkString
```

Reply:
That inconsistency is quite unfortunate.
Using the `split` to parse the path out was my initial take on the problem but it fails if the path has white spaces in it.

If you don't mind that would help a lot if you could run the script below and share the output of `gitlog` file from the `status_test` directory.

```shell
#!/usr/bin/env bash

function commit() {
  commit_message="$1"
  git add .
  echo "$commit_message" >> gitlog
  echo "-------git status-------" >> gitlog
  echo "$(git status)" >> gitlog
  echo "-------git status --porcelain-------" >> gitlog
  echo "$(git status --porcelain)" >> gitlog
  echo "-------git status --porcelain=1-------" >> gitlog
  echo "$(git status --porcelain=1)" >> gitlog
  echo "-------git status --porcelain=2-------" >> gitlog
  echo "$(git status --porcelain=2)" >> gitlog
  echo "-------git status -z-------" >> gitlog
  echo "$(git status -z)" >> gitlog
  echo "-------git status -z --porcelain=1-------" >> gitlog
  echo "$(git status -z --porcelain=1)" >> gitlog
  echo "-------git status -z --porcelain=2-------" >> gitlog
  echo "$(git status -z --porcelain=2)" >> gitlog
  echo "--------------" >> gitlog
  echo "--------------" >> gitlog
  git commit -am ${commit_message}
}

test_dir="status_test"

mkdir ${test_dir}
cd ${test_dir}
git init .
touch gitlog
commit "Init"
mkdir dir1
mkdir dir2
mkdir "dir 1"
echo "blah blah" > dir1/file1.txt
echo "blah blah" > dir2/file2.txt
echo "blah blah" > dir\ 1/file3.txt
commit "c1"

echo "blah blah" >> dir1/file1.txt
mv dir2/file2.txt  dir\ 1/file2.txt
rm dir\ 1/file3.txt
commit "c2"
```